prime-router/src/main/kotlin/history/azure/ReportFileFunction.kt

```diff
@@ -190,6 +205,85 @@ abstract class ReportFileFunction(
         }
     }
 
+    /**
+     * Function to return metadata of a single report from the CDC Intermediary.
+     * The [reportId] is a valid report UUID. This function is for the Intermediary only, please don't update
+     * without contacting that engineering team
+     */
+    fun retrieveETORIntermediaryMetadata(
+        request: HttpRequestMessage<String?>,
+        reportId: UUID,
+        context: ExecutionContext,
+        engine: HttpClientEngine?,
+    ): HttpResponseMessage {
+        val authResult = this.authSingleBlocks(request, reportId.toString())
+
+        if (authResult != null) {
+            return authResult
+        }
+
+        var response: HttpResponse?
+        val receiver = workflowEngine.settings.findReceiver(this.intermediaryReceiverName)
+        val client = if (engine == null) HttpClient() else HttpClient(engine)
+        val restTransportInfo = receiver?.transport as RESTTransportType
+        val (credential, jksCredential) = RESTTransport().getCredential(restTransportInfo, receiver)
+        val logger: Logger = context.logger
+        var authPair: Pair<Map<String, String>?, String?> =
+            Pair(null, null)
+
+        var responseBody = ""
+
+        runBlocking {
+            launch {
+                authPair = RESTTransport().getOAuthToken(
+                    restTransportInfo,
+                    reportId.toString(),
+                    jksCredential,
+                    credential,
+                    logger
+                )
+            }
+        }
+
+        val lookupId = this.getLookupId(reportId)
```

Comment:
Intellij recommends:

        val lookupId = this.getLookupId(reportId) ?: return HttpUtilities.notFoundResponse(request, "lookup Id not found")

Reply:
Fixed.